src/rpp/rpp/sources/concat.hpp

```diff
@@ -90,6 +90,7 @@ namespace rpp::details
             if (state->itr.value() == std::cend(state->container))
             {
                 state->observer.on_completed();
+                state->dispose();
```

Comment:
not sure if this one is really needed, same thing if we need a dispose call on line 109

Reply:
yeah, actually half of this calls would happen inside on_error/on_completed of downstream